src/Baseline.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BackwardCompatibility;
+
+use function array_values;
+use function preg_match;
+
+/** @psalm-immutable */
+final class Baseline
+{
+    /** @psalm-param list<string> $ignoredChanges */
+    private function __construct(private readonly array $ignoredChanges = [])
+    {
+    }
+
+    public static function empty(): self
+    {
+        return new self();
+    }
+
+    public static function fromList(string ...$ignoredChanges): self
+    {
+        return new self(array_values($ignoredChanges));
+    }
+
+    public function ignores(Change $change): bool
+    {
+        foreach ($this->ignoredChanges as $ignoredChangeRegex) {
```

Comment:
We could compile a single regex with all the items and just do one regex match per detected changes. It comes with a certain complexity, though.

Reply:
I can imagine if there are a huge amount of changes found and/or the regex list is huge, this could become a performance issue, but we can probably leave it as-is until it becomes a problem; it's understandable at least for now.

Reply:
I'd suggest leaving it as is for this iteration, honestly.

Compiling efficient regexes out of a variable list would also certainly impact performance and there's no way to guarantee we will have **one regex only**.